2-mandatory/3-financial-times.js

```diff
@@ -14,28 +20,54 @@ function potentialHeadlines(allArticleTitles) {
     (you can assume words will always be seperated by a space)
 */
 function titleWithFewestWords(allArticleTitles) {
-    // TODO
-}
+         let fewestWords = "";
+    for (nElememntInArray of allArticleTitles){
+        let titleLength = 0;
+        for (characterIntitle of nElememntInArray){
+        titleLength +=1;
+        }
+        if (nElememntInArray.length <= titleLength ) {
+         fewestWords = nElememntInArray;
+        } 
+    }
+    
+    return fewestWords;
 
+    }
+    
 /*
     The editor of the FT has realised that headlines which have numbers in them get more clicks!
     Implement the function below to return a new array containing all the headlines which contain a number.
     (Hint: remember that you can also loop through the characters of a string if you need to)
 */
 function headlinesWithNumbers(allArticleTitles) {
-    // TODO
+    let headlineOnNumber = [];
+    for (let i = 0; i < allArticleTitles.length; i++){// TODO
+        for (let loopOnElement of allArticleTitles[i]){
+            if ( loopOnElement === "$"){
```

Comment:
There is anthor simpler way of checking if a string contains number, using rejex check this out:
```function headlinesWithNumbers(allArticleTitles) {
  let NewArr = [];
  for (let index = 0; index < allArticleTitles.length; index++) {
    if (/\d/.test(allArticleTitles[index]))
      NewArr.push(allArticleTitles[index]);
  }
  return NewArr;
}```


In your case, you are checking `$` sign but that might change I mean you might have a number without `$` so .....

Reply:
yes you are right, I use the sign "$" because I ran out of options and need this code to compile. I know this is now what they  asked. 
  I did not really understand this ` if (/\d/.test(allArticleTitles[index])) `but I should have use the condition below.

            if (loopOnElement >="0"  or  loopOnElement <="9") then push